src/transport/SessionManager.cpp

```diff
@@ -486,21 +486,25 @@ void SessionManager::SecureGroupMessageDispatch(const PacketHeader & packetHeade
         ExitNow(err = CHIP_NO_ERROR);
     }
 
+    // Group Messages should never send an Ack
+    if (payloadHeader.NeedsAck())
+    {
+        ChipLogError(Inet, "Invalid condition found in protocol header");
```

Comment:
We could make the message less vague since we know exactly what went wrong. How about:

"Unexpected ACK requested for group message"

Reply:
Fair enough, changed.